### ISSUE TITLE
Support visibility/security enforcement for kafka over TLS

### ISSUE THREAD

**Issue by nnahaliel-onezero** [cilium/cilium#17166]
## Proposal / RFE

I have multiple services connecting to an external kafka service over TLS, and I want to restrict pods access to required kafka topics only.
The documentation doesn't specifically say that this is unsupported, but after asking at Cilium's slack channel, i was advised to give it a try, and it seems that if i take a working http over tls policy and change it to kafka rules and port, then this "new" policy is ignored.

I'm not sure if this is a feature request, a bug or just a misconfiguration on my part.

### General Information

- AWS
- cilium v1.8.11
- Kubernetes (EKS) v1.20

**Comment by ti-mo** [comment]
Hi @nnahaliel, community Slack is a better platform for questions like these. A proposal/RFE issue is to open a discussion if you're planning to implement a particular feature into Cilium itself.

Have you been able to set up a working HTTPS example using https://docs.cilium.io/en/v1.10/gettingstarted/tls-visibility/? Have you done any troubleshooting upon observing Kafka over TLS doesn't work?

**Comment by nnahaliel** [comment]
Hi,

> Hi @nnahaliel, community Slack is a better platform for questions like these. A proposal/RFE issue is to open a discussion if you're planning to implement a particular feature into Cilium itself.

I tried asking at the community slack. Like I wrote above, I was told it's not tested but should work. Gave it a try and couldn't make it work. I commented that fact on the thread i was on but got no response.
I wasn't sure about tagging this as a feature, but since I was told it wasn't tested and the docs don't specifically say this is supported, I thought that I can't really define it as a bug.

> Have you been able to set up a working HTTPS example using https://docs.cilium.io/en/v1.10/gettingstarted/tls-visibility/? Have you done any troubleshooting upon observing Kafka over TLS doesn't work?

Yes, After trying to build a kafka over TLS policy from scratch, I created an HTTPS policy, confirmed it worked, and then changed just the rule section (from http to kafka) and port.
I did see that that after running "cilium policy get" on a cilium pod, the "kafka-over-tls" policy was not in the output.
Before submiting this issue i did not find anything useful in the logs, but after further examination I found this (the name of the policy is - "l7-visibility-tls" because that's how I copied it from [the docs](https://docs.cilium.io/en/v1.8/gettingstarted/tls-visibility/)) - 

`level=warning msg="Unable to add CiliumNetworkPolicy" ciliumNetworkPolicyName=l7-visibility-tls error="Invalid CiliumNetworkPolicy spec: L7 protocol Kafka is not supported on egress yet" k8sApiVersion= k8sNamespace=cilium-tls-test subsys=k8s-watcher`

So I guess that might be it, although I recall defining an egress kafka policy for plaintext communication a while ago on the same cluster (now doubting my memory).
